skrub/_reporting/_utils.py

```diff
@@ -87,7 +87,10 @@ def format_number(number):
     if isinstance(number, numbers.Integral):
         return f"{number:,}"
     if isinstance(number, numbers.Real):
-        return f"{number:#.3g}"
+        from skrub._config import get_config
+
+        var = get_config()["float_precision"]
+        return f"{number:#.{var}g}"
```

Comment:
Pandas display doesn't force decimal representation, so I think we should be consistent with that. WDYT @rcap107?
```suggestion
        return f"{number:.{var}g}"
```

Reproducer:
```python
import pandas as pd
x = 123
print(f"{x:.3g}")
pd.set_option("display.precision", 3)
pd.DataFrame(dict(a=[x]))
```

Reply:
You could argue that casting integers to floats doesn't help distinguish floats from ints either. So both options are valid, and I don't have a strong pref.